.travis.yml

```diff
@@ -101,3 +101,20 @@ jobs:
           script: ./docker/deploy.sh $TRAVIS_TAG
           on:
             tags: true
+    - name: "Deploy Price Estimator"
```

Comment:
This shows one disadvantage of the monorepo: We have to deploy the price estimator every time a new commit lands in master. It wouldn't be enough to check to the last modification date of the price-estimator folder because a change in pricegraph or core could affect the priceestimator. If the price estimator was in it's own repo (for example using a git submodule for core and pricegraph or a cargo git dependency) then this wouldnt' be the case. Of course in that case we would have to manually update the dependency when needed.
I'm not saying we should do this just want to point it out.

Reply:
I think this is also a "pro" since we have CD and integration errors with `price-estimator` and `driver` with `core` and `pricegraph` are caught early and tested often. It just adds CI overhead, but Travis is a :robot: and doesn't mind :smile:.

One point though (that you've made before) that I find more of an issue is regarding deploying tagged versions. We currently have no way of individually deploying the price estimator and driver which can be annoying, especially if we need to do a hotfix.

Reply:
If we want to be fancy we can look at the project dependency graph and skip a build if no file in the dependency graph was changed. This is how large scale monorepos do it if I'm not mistaken.

Reply:
I think, as @e00E suggested, most of the code is shared anyway:

```
$ find $(ls -d */ | grep -v dex-contracts | grep -v target) -iname "*.rs" -exec cat {} \; | wc -l     
16529
$ find driver price-estimator -iname "*.rs" -exec cat {} \; | wc -l 
900
```

So a change in roughly 95% of the codebase would require building both driver and the price-estimator (note I did not include `e2e/` in the second line count as I imagine we might want to add price estimator e2e tests eventually).